src/state.rs

```diff
@@ -66,6 +67,7 @@ impl State {
 /// Readonly snapshot of Language server state.
 pub struct StateSnapshot {
     pub db: salsa::Snapshot<AnalysisDatabase>,
+    pub scarb_toolchain: Snapshot<ScarbToolchain>,
```

Comment:
```suggestion
    pub scarb_toolchain: ScarbToolchain,
```

`ScarbToolchain` is a sync singleton. We don't need `Owned`/`Snapshot` system for it imho